src/draw/lv_draw_vector.h

```diff
@@ -0,0 +1,145 @@
+/**
+ * @file lv_draw_vector.h
+ *
+ */
+
+#ifndef LV_DRAW_VECTOR_H
+#define LV_DRAW_VECTOR_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_draw.h"
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+enum {
+    LV_VECTOR_PATH_FILL_NONZERO = 0,
+    LV_VECTOR_PATH_FILL_EVENODD
+};
+typedef uint8_t lv_vector_path_fill_rule_t;
+
+enum {
+    LV_VECTOR_PATH_PATTERN_MODE_COLOR = 0,
+    LV_VECTOR_PATH_PATTERN_MODE_PAD,
+};
+typedef uint8_t lv_vector_path_pattern_mode_t;
+
+enum {
+    LV_VECTOR_STROKE_CAP_STYLE_SQUARE = 0,
+    LV_VECTOR_STROKE_CAP_STYLE_ROUND,
+    LV_VECTOR_STROKE_CAP_STYLE_BUTT,
+};
+typedef uint8_t lv_vector_stroke_cap_style_t;
+
+enum {
+    LV_VECTOR_STROKE_JOIN_STYLE_BEVEL = 0,
+    LV_VECTOR_STROKE_JOIN_STYLE_ROUND,
+    LV_VECTOR_STROKE_JOIN_STYLE_MITER,
+};
+typedef uint8_t lv_vector_stroke_join_style_t;
+
+typedef struct {
+    float m[3][3];
+} lv_matrix_t;
```

Comment:
```suggestion
typedef float (lv_matrix_t)[3][3];
```

Reply:
As per https://stackoverflow.com/a/4523537 I would like to keep current implementations, what do you think?